ledger/src/blockstore_processor.rs

```diff
@@ -110,25 +113,30 @@ fn first_err(results: &[Result<()>]) -> Result<()> {
 fn get_first_error(
     batch: &TransactionBatch<impl SVMTransaction>,
     commit_results: &[TransactionCommitResult],
+    is_block_producing_unified_scheduler: bool,
```

Comment:
It seems odd to me that we would call this function at all for a block-producing scheduler, it seems to make more sense if we had the if switch around the call than inside to me.

Reply:
Also maybe a dumb question - IF we detect an error here we discard the block as dead.
Why commit the results of such transactions at all?
Why not make this function (or slight modification of it) a pre-commit check for the block-verification variant?